core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1085,6 +1085,10 @@ class DAGScheduler(
 
         if (disallowStageRetryForTest) {
           abortStage(failedStage, "Fetch failure will not retry stage due to testing config")
+        } else if (failedStage.failAndShouldAbort()) {
```

Comment:
yeah, sucks we are discovering this so close to the release.  In any case, I opened https://issues.apache.org/jira/browse/SPARK-7308 , we can discuss in more detail there.